microsoft/testsuites/vm_extensions/azure_keyvault_vm_extension.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import os
+import random
+import time
+from typing import List
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.base_tools.service import Service
+from lisa.operating_system import BSD
+from lisa.sut_orchestrator.azure.common import (
+    add_system_assign_identity,
+    assign_access_policy_to_vm,
+    check_certificate_existence,
+    create_certificate,
+    create_keyvault,
+    delete_certificate,
+    delete_keyvault,
+    get_node_context,
+    rotate_certificates,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.testsuite import TestResult
+from lisa.tools.ls import Ls
+from lisa.util import LisaException
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="BVT for Azure Key Vault Extension",
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class AzureKeyVaultExtensionBvt(TestSuite):
+    # Private method for checking system status
+    def _check_system_status(self, node: Node, log: Logger) -> None:
```

Comment:
Move private methods behind the public methods. For this method, it doesn't use any method of the test class, so it can be a function, instead of a method in the class.